its/src/org/sonarlint/eclipse/its/SonarQubeConnectedModeTest.java

```diff
@@ -174,10 +206,55 @@ public void testLocalServerStatusRequest() throws Exception {
   }
 
   @Test
-  public void shouldFindSecretsInConnectedMode() throws Exception {
+  public void shouldFindSecretsInConnectedMode() {
     new JavaPerspective().open();
     var rootProject = importExistingProjectIntoWorkspace("secrets/secret-java", PROJECT_NAME);
 
+    createConnectionAndBindProject(PROJECT_NAME);
+
+    openFileAndWaitForAnalysisCompletion(rootProject.getResource("src", "sec", "Secret.java"));
+
+    var defaultEditor = new DefaultEditor();
+    assertThat(defaultEditor.getMarkers())
+      .extracting(Marker::getText, Marker::getLineNumber)
+      .containsOnly(
+        tuple("Make sure this AWS Secret Access Key is not disclosed.", 4));
+
+    var preferencesShell = new DefaultShell("SonarLint - Secret(s) detected");
+    new DefaultLink(preferencesShell, "Dismiss").click();
+  }
+
+  @Test
+  public void shouldAutomaticallyUpdateRuleSetWhenChangedOnServer() throws Exception {
+    Assume.assumeTrue(orchestrator.getServer().version().isGreaterThanOrEquals(9, 4));
+
+    new JavaPerspective().open();
+    var rootProject = importExistingProjectIntoWorkspace("java/java-simple", JAVA_SIMPLE_PROJECT_KEY);
+
+    createConnectionAndBindProject(JAVA_SIMPLE_PROJECT_KEY);
+
+    var file = rootProject.getResource("src", "hello", "Hello.java");
+    openFileAndWaitForAnalysisCompletion(file);
+
+    var defaultEditor = new TextEditor();
+    assertThat(defaultEditor.getMarkers())
+      .extracting(Marker::getText, Marker::getLineNumber)
+      .containsOnly(
+        tuple("Replace this use of System.out or System.err by a logger.", 9));
+
+    var qualityProfile = getQualityProfile(JAVA_SIMPLE_PROJECT_KEY, "SonarLint IT Java");
+    deactivateRule(qualityProfile, "S106");
+    Thread.sleep(5000);
+
+    doAndWaitForSonarLintAnalysisJob(() -> {
```

Comment:
Would be great at some point to have analysis automatically triggered when rule is changed.

Reply:
We haven't done it on SLI either, we could consider it later yes